backend/src/register/register.controller.ts

```diff
@@ -1,21 +1,22 @@
 import { Body, Controller, Get, Param, Post } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
 import { RegisterDto } from '../dto/register.dto';
 import { RegisterService } from './register.service';
 
 @Controller()
 export class RegisterController {
-    constructor(private readonly registerService: RegisterService) {}
+    constructor(private readonly registerService: RegisterService, private readonly configService: ConfigService) {}
 
     @Post('register')
-    async registerBadge(@Body() { did, signature, service }: RegisterDto) {
+    async registerProfile(@Body() { did, signature, service }: RegisterDto) {
         const network = did.match(/^did:ethr:(.+):/)?.[1];
 
         const txHash = await this.registerService.addService(
             did,
             network,
             {
                 type: 'test',
-                serviceEndpoint: 'https://ipfs.io/ipfs/ID',
+                serviceEndpoint: this.configService.get('SERVICE_ENDPOINT') + signature,
```

Comment:
You can use the "service" argument to help provide the values:
{
  type: service.type,
  serviceEndpoint: service. serviceEndpoint
}

But since the server knows the IPFS location and the client doesn't. One idea:  pass the ipfs-hash from client. We can change the RegisterDto.RegisterServiceDto to include the IPFS hash then combine them on the server:

serviceEndpoint: this.configService.get('SERVICE_ENDPOINT') + service.cid

Reply:
I changed: **type: 'test** into **type: service.type**
I'd say that signature already contains that hash. It looks something like: 
0xc9618048d4e98b985b661f896b...

Reply:
That is the transaction client signature. Use the ipfs-hash. It will look like QmcYGhqKc8LgXo6WTZbbNJtTR6...

Reply:
You can check it now.